org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JVMConfigurator.java

```diff
@@ -108,6 +114,10 @@ public static boolean configureJVMs(Preferences preferences, JavaClientConnectio
 					IPath sourcePath = runtime.getSourcePath();
 					IVMInstall vm = findVM(file, runtime.getName());
 					IVMInstallType installType = JavaRuntime.getVMInstallType(StandardVMType.ID_STANDARD_VM_TYPE);
+					if (installType == null || installType.getVMInstalls().length == 0) {
+						// https://github.com/eclipse/eclipse.jdt.ls/issues/1646
+						installType = JavaRuntime.getVMInstallType(MAC_OSX_VM_TYPE);
```

Comment:
When i'm using the launch file `jdt.ls.socket-stream.launch` to launch the eclipse.jdt.ls, it will throw NPE. The reason is `org.eclipse.jdt.launching.macosx` is not added to the required bundles list, this new line will assign null to installType.

We should check if `JavaRuntime.getVMInstallType(MAC_OSX_VM_TYPE)` is null. We can use it only when it's not null.

Reply:
I have created https://github.com/eclipse/eclipse.jdt.ls/issues/1696